corpus/filter.py

```diff
@@ -321,21 +325,50 @@ def unknown_filter(corpus: corpus.Corpus, recording: corpus.Recording, segment:
                 return all(w in vocabulary for w in words)
 
         c.filter_segments(unknown_filter)
+
+        if self.delete_empty_recordings:
+            # Remove the recordings without segments due to the filtering.
+            to_delete = []
+            for rec in c.all_recordings():
+                if not rec.segments:
+                    to_delete.append(rec)
+
+                for rec in to_delete:
+                    c.remove_recording(rec)
+                with open(self.out_removed_recordings, "w") as f:
+                    f.write("\n".join(rec.fullname() for rec in to_delete))
+
         c.dump(self.out_corpus.get_path())
 
 
 class FilterCorpusBySegmentDurationJob(Job):
-    def __init__(self, bliss_corpus: Path, min_duration: float = 0.1, max_duration: float = 120.0):
+    """
+    Removes all segments from all corpus recordings that don't fall within the specified duration boundaries.
+    """
+
+    __sis_hash_exclude__ = {"delete_empty_recordings": False}
+
+    def __init__(
+        self,
+        bliss_corpus: Path,
+        min_duration: float = 0.1,
+        max_duration: float = 120.0,
+        delete_empty_recordings: bool = True,
```

Comment:
```suggestion
        delete_empty_recordings: bool = False,
```
while i agree that `True` should be used by default, if we change the actual default value to `True`, then this breaks the hashes of all existing setups that used this job and did not specify a value.

Reply:
Thanks, I meant to set to False for hash compatibility purposes but this slipped past me.

Reply:
As you see, the AppTek hashes pipeline had also identified this problem. 👍